### DISCUSSION TITLE
Run transform conditionally in aggregator model

### DISCUSSION THREAD

**Discussion by olivermussell** [vectordotdev/vector#19380] [Q&A]
We're running Vector as a sidecar in AWS Fargate/ECS to collect logs from containers and then send them via a Vector aggregator to Elasticsearch. 

I need to run a transform to amend the log structure from some of our containers and currently do this in the agent/sidecar config.toml. However, its not clear whether this is the correct approach in the aggregator model and if it would instead be better to do this in the aggregator config. 

So is it possible to move the transform into the aggregator config with an if statement? We would be sending logs from many containers but only need to run the transform on logs from some specific containers so I wondered if there might be some way to say "if log contains this thing, run this transform". The only place I've seen in the docs is in a filter, but we don't want to drop the logs but run a transform instead.

Or otherwise is it preferred to transform the data in the agent before it is forwarded to the aggregator?

For reference:
- [agent.toml](https://gitlab.com/crossref/infrastructure/observability/-/blob/main/agent.toml?ref_type=heads)
- [aggregator.toml](https://gitlab.com/crossref/infrastructure/observability/-/blob/main/aggregator.toml?ref_type=heads)

**Comment by jszwedko** [comment]
The equivalent of an "if statement" would be to use a `route` transform in the aggregator to split the events and only run the conditional transform on a subset of the events, and then have the downstream component consume both input streams.
